File: tests/python/password_encoder.py
"""
Utility for encoding and validating passwords.
"""
import hashlib
import base64


class PasswordEncoder:
    """
    Utility for encoding and validating passwords.
    Uses SHA-256 hashing for demonstration (use bcrypt in production).
    """

    @staticmethod
    def encode(plain_password: str) -> str:
        """Encode a plain text password."""
        hash_obj = hashlib.sha256(plain_password.encode())
        return base64.b64encode(hash_obj.digest()).decode()

    @staticmethod
    def matches(plain_password: str, encoded_password: str) -> bool:
        """Check if a plain password matches an encoded password."""
        encoded = PasswordEncoder.encode(plain_password)
        return encoded == encoded_password

File: tests/javascript/PasswordEncoder.js
const crypto = require('crypto');

/**
 * Utility for encoding and validating passwords.
 * Uses SHA-256 hashing for demonstration (use bcrypt in production).
 */
class PasswordEncoder {
  /**
   * Encode a plain text password.
   * @param {string} plainPassword - The password to encode
   * @returns {string} The encoded password
   */
  encode(plainPassword) {
    const hash = crypto.createHash('sha256');
    hash.update(plainPassword);
    return hash.digest('base64');
  }

  /**
   * Check if a plain password matches an encoded password.
   * @param {string} plainPassword - The plain text password
   * @param {string} encodedPassword - The encoded password
   * @returns {boolean} True if passwords match
   */
  matches(plainPassword, encodedPassword) {
    const encoded = this.encode(plainPassword);
    return encoded === encodedPassword;
  }
}

module.exports = PasswordEncoder;
